src/components/fields/password-field/password-field.js

```diff
@@ -18,127 +18,105 @@ const sequentialId = createSequentialId('password-field-');
 
 const hasErrors = errors => errors && Object.values(errors).some(Boolean);
 
-class PasswordField extends React.Component {
-  static displayName = 'PasswordField';
+const PasswordField = props => {
+  const intl = useIntl();
+  const [isPasswordVisible, setIsPasswordVisible] = React.useState(false);
+  const id = getFieldId(props, {}, sequentialId);
+  const hasError = props.touched && hasErrors(props.errors);
 
-  static propTypes = {
-    // PasswordField
-    id: PropTypes.string,
-    horizontalConstraint: PropTypes.oneOf(['s', 'm', 'l', 'xl', 'scale']),
-    errors: PropTypes.shape({
-      missing: PropTypes.bool,
-    }),
-    renderError: PropTypes.func,
-    isRequired: PropTypes.bool,
-    touched: PropTypes.bool,
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-
-    // PasswordInput
-    name: PropTypes.string,
-    value: PropTypes.string.isRequired,
-    onChange: requiredIf(PropTypes.func, props => !props.isReadOnly),
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    isAutofocussed: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    isReadOnly: PropTypes.bool,
-    placeholder: PropTypes.string,
-    autoComplete: PropTypes.string,
+  return (
+    <Constraints.Horizontal constraint={props.horizontalConstraint}>
+      <Spacings.Stack scale="xs">
+        <Spacings.Inline alignItems="center" justifyContent="space-between">
+          <FieldLabel
+            hint={props.hint}
+            title={props.title}
+            badge={props.badge}
+            htmlFor={id}
+            hintIcon={props.hintIcon}
+            description={props.description}
+            onInfoButtonClick={props.onInfoButtonClick}
+            hasRequiredIndicator={props.isRequired}
+          />
+          {!props.isDisabled && !props.isReadOnly && (
+            <FlatButton
+              icon={isPasswordVisible ? <EyeCrossedIcon /> : <EyeIcon />}
+              label={
+                isPasswordVisible
+                  ? intl.formatMessage(messages.hide)
+                  : intl.formatMessage(messages.show)
+              }
+              onClick={() => setIsPasswordVisible(!isPasswordVisible)}
```

Comment:
Nit: could be a tiny memoized fn togglePasswordVisibility.

Reply:
is this what you mean?

https://github.com/commercetools/ui-kit/pull/957/commits/4200a7b358180cd3aed72d996a1ed08e277e22c7

I'm not a huge hook expert yet 🤓